### PR TITLE
Switch to gorilla WebSockets.

### PR DESCRIPTION
**Pull request by frankban**
This way we add support for the chunked messages sent by Juju.

Also detect local addresses and suggest correct URLs for visiting the GUI.

### CODE REVIEW

File: internal/network/network.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package network
+
+import "net"
+
+// Addresses returns the list of addresses for the local machine.
+// If ipv4 addresses are available, those are preferred over ipv6 ones.
+func Addresses() ([]string, error) {
```
Comment (hatched) [inline internal/network/network.go:7]: `GetAddresses`? 🤷‍♂️

File: server/server.go [hunk 2]
```diff
@@ -36,21 +36,24 @@ const (
 	// dynamically generated Juju GUI configuration file.
 	jujuVersion       = "2.2.0"
 	legacyJujuVersion = "1.25.7"
+
+	// webSocketBufferSize holds the frame size for WebSocket messages.
+	webSocketBufferSize = 65536
```
Comment (hatched) [inline server/server.go:41]: Does this have to be kept in sync with Juju? Might be worth a comment to that effect.

**Review by hatched — APPROVED** [review]
Thanks @frankban this will sure be nice to avoid those random disconnections :)

File: internal/network/network.go (hunk 1 above)
Comment (frankban) [reply internal/network/network.go:7]: Without the Get feels more Go idiomatic, like net.Interfaces(), strings.Fields() etc.

File: server/server.go (hunk 2 above)
Comment (frankban) [reply server/server.go:41]: No it doesn't. It can go out of sync without problems.

**Comment by hatched** [comment]
Q A O K !

**Review by makyo — APPROVED** [review]
Looks good; Gorilla is much cleaner.

**Pull request merged**